src/Console/ConfigurationResolver.php

```diff
@@ -702,16 +703,85 @@ private function validateRules(array $rules): void
+                ],
+                'lowercase_constants' => [
+                    'new_name' => 'constant_case',
+                    'config' => ['case' => 'lower'],
+                ],
+                'no_extra_consecutive_blank_lines' => [
+                    'new_name' => 'no_extra_blank_lines',
+                    'config' => null,
+                ],
+                'no_multiline_whitespace_before_semicolons' => [
+                    'new_name' => 'multiline_whitespace_before_semicolons',
+                    'config' => null,
+                ],
+                'no_short_echo_tag' => [
+                    'new_name' => 'echo_tag_syntax',
+                    'config' => ['format' => 'long'],
+                ],
+                'php_unit_ordered_covers' => [
+                    'new_name' => 'phpdoc_order_by_value',
+                    'config' => ['annotations' => ['covers']],
+                ],
+                'phpdoc_inline_tag' => [
+                    'new_name' => 'general_phpdoc_tag_rename, phpdoc_inline_tag_normalizer and phpdoc_tag_type',
+                    'config' => null,
+                ],
+                'pre_increment' => [
+                    'new_name' => 'increment_style',
+                    'config' => ['style' => 'pre'],
+                ],
+                'psr0' => [
+                    'new_name' => 'psr_autoloading',
+                    'config' => ['dir' => 'x'],
+                ],
+                'psr4' => [
+                    'new_name' => 'psr_autoloading',
+                    'config' => null,
+                ],
+                'silenced_deprecation_error' => [
+                    'new_name' => 'error_suppression',
+                    'config' => null,
+                ],
+                'trailing_comma_in_multiline_array' => [
+                    'new_name' => 'trailing_comma_in_multiline',
+                    'config' => ['elements' => ['arrays']],
+                ],
+            ];
+
             $matcher = new WordMatcher($availableFixers);
 
             $message = 'The rules contain unknown fixers: ';
             foreach ($unknownFixers as $unknownFixer) {
-                $alternative = $matcher->match($unknownFixer);
-                $message .= sprintf(
-                    '"%s"%s, ',
-                    $unknownFixer,
-                    null === $alternative ? '' : ' (did you mean "'.$alternative.'"?)'
-                );
+                // Check if present as old renamed rule
+                if (isset($renamedRules[$unknownFixer])) {
+                    $message .= sprintf(
+                        '"%s" is a renamed rule, %s, ',
+                        $unknownFixer,
+                        ' did you mean "'.$renamedRules[$unknownFixer]['new_name'].'"'.(!empty($renamedRules[$unknownFixer]['config']) ? ' (note: use configuration "'.HelpCommand::toString($renamedRules[$unknownFixer]['config']).'")' : '').'?'.PHP_EOL.PHP_EOL.
```

Comment:
please simplify the `sprintf` :)

```
$message .= sprintf(
    "\"%s\" is a renamed rule, did you mean \"%s\"%s?\nFor more info see: https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v3.0.0/UPGRADE-v3.md#renamed-rules",
    $unknownFixer,
    $renamedRules[$unknownFixer]['new_name'],
    null !== $renamedRules[$unknownFixer]['config'] ? ' (note: use configuration "'.HelpCommand::toString($renamedRules[$unknownFixer]['config']).'")' : '')
);
```

Reply:
just a little thingy from me and than I think this is RTM for me :+1:  thanks for taking the time to put this together :)

Reply:
Done in https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5887/commits/ff002a76a1ffb3f9216130bf8205f69bf25e0ecc